web/packages/teleterm/src/ui/components/OfflineGateway.tsx

```diff
@@ -136,8 +133,14 @@ export function OfflineGateway(props: {
   );
 }
 
-export const FIELD_NAME_LOCAL_PORT = 'localPort';
+export enum FormFields {
+  LocalPort = 'localPort',
+  TargetSubresourceName = 'targetSubresourceName',
+}
+type FormFieldNames = `${FormFields}`;
```

Comment:
Magic

Reply:
Yeah, too much magic. 😩 But I wanted to use string enums here and this seems to be the most succinct way of getting the actual values out of the enum.